hardware/serial/serial.cpp

```diff
@@ -2,7 +2,7 @@
 #include "stdafx.h"
 #include <algorithm>
 
-#if !defined(_WIN32) && !defined(__OpenBSD__) && !defined(__FreeBSD__)
+#if !defined(_WIN32) && !defined(__OpenBSD__) && !defined(__FreeBSD__) && !defined(__NetBSD__)
 # include <alloca.h>
```

Comment:
We can reverse this and include `<alloca,h>` for `__sun` only. Everybody else can include `<stdlib.h>`.

Reply:
Alternatively we can detect alloca.h with cmake.

Reply:
That does look nicer. As for detecting stuff with cmake, I'll take your word for it. :)